src/Compilers/CSharp/Portable/CodeGen/EmitExpression.cs

```diff
@@ -871,12 +871,21 @@ private void EmitFieldLoad(BoundFieldAccess fieldAccess, bool used)
         {
             var field = fieldAccess.FieldSymbol;
 
-            //TODO: For static field access this may require ..ctor to run. Is this a side-effect?
-            // Accessing unused instance field on a struct is a noop. Just emit the receiver.
-            if (!used && !field.IsVolatile && !field.IsStatic && fieldAccess.ReceiverOpt.Type.IsVerifierValue())
+            if (!used)
             {
-                EmitExpression(fieldAccess.ReceiverOpt, used: false);
-                return;
+                // fetching unused captured frame is a no-op (like reading "this")
+                if(field.IsCapturedFrame)
+                {
+                    return;
+                }
+
+                //TODO: For static field access this may require ..ctor to run. Is this a side-effect?
```

Comment:
I realize this is an old comment but we should take the time to correct.  If this is a real bug please file an issue for the TODO and reference it here.  Otherwise if it's not a bug let's clean up this comment.